pkg/trace/agent/agent.go

```diff
@@ -536,18 +536,29 @@ func traceContainsError(trace pb.Trace) bool {
 	return false
 }
 
-func filteredByTags(root *pb.Span, require, reject []*config.Tag) bool {
+func filteredByTags(root *pb.Span, require, reject []*config.Tag, requireRegex, rejectRegex []*config.TagRegex) bool {
```

Comment:
I think it's worth adding a benchmark here for spans with many tags. If it's significant (which it likely will be, considering it's regexp), then we may want to note in the docs that this could have performance implications.

Reply:
Unfortunately, I no longer participate in the project in which the datadog agent was used. Therefore, I have no way to assess how much the use of regexp for span filtering reduces the tracking performance.

Reply:
I just meant a [Go benchmark ](https://pkg.go.dev/testing#hdr-Benchmarks) for this function with and without `rejectRegex` and `requireRegex` set. I assume that running a regular expression on every tag of each root span could have performance implications. But hopefully that doesn't need to be explicitly stated in the release notes, and users will understand that without it being stated outright. Fortunately this is only run on the root span of a trace, not every single span, so that should make this a bit less painful. 

If you don't feel comfortable adding a benchmark or it's not feasible, just let me know!

Reply:
That's it, now I understand what benchmark we are talking about.
I added a benchmark.

Here are the Benchmark launch results:
```shell
BenchmarkFilteredByTags/FilteredByTags-10         	46732989	        25.03 ns/op
BenchmarkFilteredByTags/FilteredByRegexTags-10    	14361446	        83.38 ns/op
```

I also add information about performance degradation when using regex
patterns to the description of the `filter_tags_regex` parameter.

Reply:
@katiehockman, If it doesn't bother you, will you be able to write your opinion about the latest changes. After that we will be able to mark this conversation as resolve.

Reply:
Looks great to me, thanks. I'll resolve this one.